src/WebJobs.Script.WebHost/WebJobsScriptHostService.cs

```diff
@@ -82,7 +81,21 @@ private enum JobHostStartupMode
             HandlingInitializationError = 16 | HandlingNonTransientError
         }
 
-        public IServiceProvider Services => _host?.Services;
+        private IHost ActiveHost
+        {
+            get
+            {
+                return _host;
+            }
+
+            set
+            {
+                _logger?.ActiveHostChanging(GetHostInstanceId(_host), GetHostInstanceId(value));
```

Comment:
Wanted to be more verbose about when we switch active hosts. That would have made the incident I investigated a little easier to reason through.